source/Background/Controller.js

```diff
@@ -323,72 +330,55 @@ backgroundController.exposeController(
     const { canisterId, requestType, preApprove } = requestInfo;
 
     try {
-      getApps(keyring.currentWalletId.toString(), async (apps = {}) => {
```

Comment:
@aleacevedo plz take a look at this. I changed a bit the check we do to validate if it's a call or not.